### ISSUE TITLE
SQLite unable to parse CREATE TABLE .. DEFAULT with parentheses

### ISSUE THREAD

**Issue by kgutwin** [sqlfluff/sqlfluff#4552]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When parsing a CREATE TABLE expression which has a column definition including a DEFAULT statement, if the default value is bracketed in parentheses, the statement is determined to be unparsable by sqlfluff.

Parentheses are optional for default literal values, but required for defaults that use functions like `datetime()` or `randomblob()`.

```
sqlite> CREATE TABLE foo (id TEXT DEFAULT datetime('now'));
Error: in prepare, near "(": syntax error (1)
sqlite> CREATE TABLE foo (id TEXT DEFAULT (datetime('now')));
sqlite> -- succeeded
```

### Expected Behaviour

```
All Finished 📜 🎉!                                                                                               
```

### Observed Behaviour

```
== [stdin] FAIL                                                                                                                                  
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: "CREATE
                       | TABLE foo (id TEXT DEFAULT (''))"
```

### How to reproduce

```
echo "CREATE TABLE foo (id TEXT DEFAULT (''));" | sqlfluff lint --dialect sqlite - 
```

Removing the parentheses (changing `('')` to `''`) restores the expected behavior.

### Dialect

sqlite

### Version

Both sqlfluff version 2.0.0 and 1.4.5 are affected.
Python version: 3.11.2

### Configuration

No configuration

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by james-johnston-thumbtack** [comment]
This is also happening with the postgres dialect.  Here is an example postgres snippet from a longer file:
```sql
CREATE TABLE tbl_test (
    row_time BIGINT NOT NULL DEFAULT (5)
);
```
The error is:
```
L:  36 | P:   1 |  PRS | Line 36, Position 1: Found unparsable section: 'CREATE
                       | TABLE tbl_test (\n    row_time BIG...'
```

**Comment by kgutwin** [comment]
Unfortunately this issue is not yet completed; while it was fixed for the postgres dialect, it was not fixed for the sqlite dialect.

```
% echo "CREATE TABLE foo (id TEXT DEFAULT (''));" | sqlfluff lint --dialect postgres -
All Finished 📜 🎉! 
% echo "CREATE TABLE foo (id TEXT DEFAULT (''));" | sqlfluff lint --dialect sqlite -
== [stdin] FAIL                                                                                                                          
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: "CREATE
                       | TABLE foo (id TEXT DEFAULT (''));..."
WARNING: Parsing errors found and dialect is set to 'sqlite'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

**Comment by Darkle** [comment]
Could this be re-opened for the sqlite issue since its not fixed yet?
